### DISCUSSION TITLE
Cypress alias does not work on Github CI?

### DISCUSSION THREAD

**Discussion by hshoja** [cypress-io/cypress#21100] [CI setup] [answered]
I have a problem in github ci, it cannot find the alias, and think it even doesn't define that, but all is well on local. I tested on both cypress:open and cypress:run

this is the command I defiend:

```js
Cypress.Commands.add("byPassLogin", () => {
  const url = Cypress.env("api_url");
  const token = "...";

  cy.saveToLocalStorage("auth_token", token);

  cy.intercept("POST", url, (req) => {
    if (req.body.operationName === "me") {
      req.reply({
        statusCode: 200,
        body: {
          data: {
            me: { id: "1", email: "test@email.com" },
          },
        },
      });
    }
  }).as("byPassLogin");
});
```

and then I used it on `beforeEach` like this

```js
describe("test account functionality", () => {
  const URL = Cypress.env("api_url");

  beforeEach(() => {
    cy.visit("/");
    cy.byPassLogin();
  });

  it.only("should logout when click on nav bar", () => {
    cy.intercept("POST", URL, (req) => {
      if (req.body.operationName === "signOut") {
        req.reply({
          statusCode: 200,
          body: {
            data: { updateUser: { errors: null, user: { id: "1" } } },
          },
        });
      }
    }).as("signOut");

    cy.wait("@byPassLogin").then(() => {
      cy.url().should("include", "/app");
      cy.get("#account").click();
      cy.get("#logout").click();
      cy.wait("@signOut").then(() => {
        cy.url().should("include", "/login");
      });
    });
  });
});


```
error on github
```
CypressError: Timed out retrying after 5000ms: cy.wait() timed out waiting 5000ms for the 1st request to the route: byPassLogin. No request ever occurred. 
```

any help would be appreciated 
cypress version: 8.4.1
react: 18
#

**Comment by muratkeremozcan** [comment]
This is not possible to reproduce without a repository that can be shared.

Effectively, aliases which is something isolated to the runner works, but when the runner is in CI it does not. This is highly unlikely and we have to see it not working to help.

**Reply by hshoja** [reply]
> This is not possible to reproduce without a repository that can be shared.
> 
> Effectively, aliases which is something isolated to the runner works, but when the runner is in CI it does not. This is highly unlikely and we have to see it not working to help.

Understand, but it's a bit hard to create a repo for that, I will do that If I didn't find anything. I also added the github error on post.

**Reply by hshoja** [reply]
> This is not possible to reproduce without a repository that can be shared.
> 
> Effectively, aliases which is something isolated to the runner works, but when the runner is in CI it does not. This is highly unlikely and we have to see it not working to help.


I also used another approach, it works on local but still not work on CI

```js
Cypress.Commands.add("byPassLogin", () => {
  const url = Cypress.env("api_url");
  const token = "...";

  cy.intercept("POST", url, (req) => {
    req.reply({
      statusCode: 200,
      body: {
        data: {
          login: { user: { id: "1", email: "test@email.com" }, token },
        },
      },
    });
  }).as("byPassLogin");

  cy.visit("/").then(() => {
    cy.get("#email").type("test@email.com");
    cy.get("#password").type("123456");
    cy.get("button[type=submit]").click();

    cy.wait("@byPassLogin").then(() => {
      cy.url().should("include", "/app");
    });
  });
```

and used it like this
```js
describe("test account functionality", () => {
  const URL = Cypress.env("api_url");

  beforeEach(() => {
    cy.byPassLogin();
  });

  it.only("should logout when click on nav bar", () => {
    cy.intercept("POST", URL, (req) => {
      if (req.body.operationName === "signOut") {
        req.reply({
          statusCode: 200,
          body: {
            data: { updateUser: { errors: null, user: { id: "1" } } },
          },
        });
      }
    }).as("signOut");

    cy.get("#account").click();
    cy.get("#logout").click();
    cy.wait("@signOut").then(() => {
      cy.url().should("include", "/login");
    });
  });
```

**Comment by hshoja** [comment, marked as answer]
Thanks @muratkeremozcan 

I succeed to manage the issue, to finding the issue I used a workflow similar to [this]( https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-recording.yml) to record my testing in `dashboard.cypress.io`, then I found it does not post form data to the correct endpoint URL, actually since the URL has been defined in env I needed to define it in CI as well. 

That's it. 
It was working since I had the env in local.
